pkg/controller/utils/conditions.go

```diff
@@ -9,6 +9,9 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// =====
+// TODO This entire UpdateConditionCheck Block could probably be refactored into the structs by way of it's own package.
+
```

Comment:
Most of this file looks like it is a great candidate to be refactored into the structs themselves, but I felt like that is too much for this PR, and each of the structs would probably be it's own card.